cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb

```diff
@@ -44,35 +85,51 @@ def issue_jwt_token_unkown_kid(token_body, algorithm = Algorithms::RS256)
   def issue_jwt_token_not_memoized_key(token_body, algorithm = Algorithms::RS256)
     @jwt_token = JWT.encode(
       token_body,
-      non_memoized_rsa_key,
+      new_rsa_key,
       algorithm,
-      { kid: jwk.kid }
+      { kid: jwk_set[default_file_name].kid }
     )
   end
 
   def jwt_token
     @jwt_token
   end
 
-  def jwk
-    @jwk ||= JWT::JWK.new(rsa_key)
+  def add_new_file(file_name)
+    add_rsa_key_to_set(file_name)
+    add_jwk_to_set(file_name)
+  end
+
+  def add_jwk_to_set(file_name)
+    jwk_set[file_name] = JWT::JWK.new(rsa_keys[file_name])
   end
 
-  def rsa_key
-    @rsa_key ||= OpenSSL::PKey::RSA.new(BITS_2048)
+  def jwk_set
+    @jwk_set ||= {}
   end
 
-  def non_memoized_rsa_key
+  def add_rsa_key_to_set(file_name)
+    rsa_keys[file_name] = new_rsa_key
+  end
+
+  def rsa_keys
+    @rsa_keys ||= {}
+  end
+
+  def new_rsa_key
     OpenSSL::PKey::RSA.new(BITS_2048)
   end
 
+  def default_file_name
+    @default_file_name ||= ""
+  end
+
```

Comment:
```suggestion
```
There's no need for empty valued default.
Init function will set the first file name as a default.